src/lib/Microsoft.Health.Events/EventConsumers/Service/EventBatchingService.cs

```diff
@@ -263,6 +263,8 @@ private void LogDataFreshness(string partitionId, string triggerReason, IEnumera
             // If no events were flushed for the partition (eg: trigger reason is ThresholdWaitReached - due to receival of MaxTimeEvent), then use the current timestamp.
             var eventTimestampLastProcessed = events?.Any() ?? false ? events.Last().EnqueuedTime.UtcDateTime : DateTime.UtcNow;
             _logger.LogMetric(EventMetrics.EventTimestampLastProcessedPerPartition(partitionId, triggerReason), double.Parse(eventTimestampLastProcessed.ToString("yyyyMMddHHmmss")));
+
+            _logger.LogMetric(EventMetrics.EventFreshnessDelayPerPartition(partitionId, triggerReason), DateTime.UtcNow.Subtract(eventTimestampLastProcessed).TotalMinutes);
```

Comment:
As discussed offline, we need to change where this gets emitted or add another point of emission for when the events fail to process.

Reply:
Further discussed offline; will add point of emission for failed processing case in second iteration.